lib/analytics-helper/test/analytics-helper.test.js

```diff
@@ -109,4 +109,34 @@ describe('Analytics helper', function () {
       done();
     });
   });
+  describe('Add analytics Impression objects', function () {
+    it('Should add a new impression if there is no previous item impression', function (done) {
+      const item = fixtures.package.item;
+      const dataLayer = [];
+      const impressionsTimestamp = {};
+      analyticsHelper.addAnalyticsImpression(item, dataLayer, impressionsTimestamp);
+      expect(dataLayer).to.have.length(1);
+      done();
+    });
+    it('Should not add a new impression if there is already one within the last 30 seconds', function (done) {
+      const item = fixtures.package.item;
+      const dataLayer = [];
+      const date10secsago = (new Date()).getTime() - 10000;
```

Comment:
@Kumjami you can also write `Date.now()` to get the timestamp in milliseconds.
see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now 😉